dragonfly/implementations/psyllid_provider.py

```diff
@@ -233,51 +238,167 @@ def get_number_of_streams(self, channel):
+        self.provider.set(self.queue_dict[channel]+request, n_triggers)
+        logger.info('Setting psyllid n-trigger/skip-tolerance to {}'.format(n_triggers))
+
+
+    def _get_n_triggers(self, channel='a'):
+        request = '.active-config.{}.eb.n-triggers'.format(str(self.channel_dict[channel]))
+        n_triggers = self.provider.get(self.queue_dict[channel]+request)['n-triggers']
+        return n_triggers
+
+
+    # tell psyllid to record a frequency mask, write it to a json file and prepare for triggered run
+    def make_trigger_mask(self, channel='a', filename='~/fmt_mask.json'):
+        if self.mode_dict[channel] != 'triggered':
+            logger.error('Psyllid instance is not in triggered mode')
+            raise core.exceptions.DriplineGenericDAQError("Psyllid instance is not in triggered mode")
 
-        logger.info('freq-only')
-        request = 'run-daq-cmd.ch'+str(self.channel_dict[channel])+'.tfrr.freq-only'
+        logger.info('Switch tf_roach_receiver to freq-only')
+        request = 'run-daq-cmd.{}.tfrr.freq-only'.format(str(self.channel_dict[channel]))
         result = self.provider.cmd(self.queue_dict[channel],request)
 
-        logger.info('update-mask')
-        request = 'run-daq-cmd.ch'+str(self.channel_dict[channel])+'.fmt.update-mask'
+        logger.info('Switch frequency_mask_trigger to update-mask')
+        request = 'run-daq-cmd.{}.fmt.update-mask'.format(str(self.channel_dict[channel]))
         result = self.provider.cmd(self.queue_dict[channel],request)
         time.sleep(1)
 
-        logger.info('run')
-        request = 'start-run'
-        result = self.provider.cmd(self.queue_dict[channel], request)
+        logger.info('Start short run to record mask')
+        self.start_run(channel ,1000, '/home/project8/fmt_update.egg')
         time.sleep(1)
 
-        logger.info('write-mask')
-        request = 'run-daq-cmd.ch'+str(self.channel_dict[channel])+'.fmt.write-mask'
+        logger.info('Write mask to file')
+        request = 'run-daq-cmd.{}.fmt.write-mask'.format(str(self.channel_dict[channel]))
         payload = {'filename': filename}
         result = self.provider.cmd(self.queue_dict[channel], request, payload=payload)
 
-        logger.info('back to time and freq')
-        request = 'run-daq-cmd.ch'+str(self.channel_dict[channel])+'.tfrr.time-and-freq'
+        logger.info('Switch tf_roach_receiver back to time and freq')
+        request = 'run-daq-cmd.{}.tfrr.time-and-freq'.format(str(self.channel_dict[channel]))
         result = self.provider.cmd(self.queue_dict[channel],request)
 
-        logger.info('apply-trigger')
-        request = 'run-daq-cmd.ch'+str(self.channel_dict[channel])+'.fmt.apply-trigger'
+        logger.info('Switch frequency mask trigger to apply-trigger')
+        request = 'run-daq-cmd.{}.fmt.apply-trigger'.format(str(self.channel_dict[channel]))
         result = self.provider.cmd(self.queue_dict[channel],request)
-        return True 
```

Comment:
Can we return something like "Awesome, the mask has been generated"?

Reply:
ok :)

Reply:
As above, I disagree with having non-informative return values. We should be consistent with the rest of the code. Unless there is a specific reason we want to try to do something and keep going if it fails, then we should raise exceptions. If such a condition exists then the function name should reflect that. Something like `make_trigger_mask_and_check()` or `check_make_trigger_mask()`.